ROSCO/src/Controllers.f90

```diff
@@ -374,6 +387,68 @@ SUBROUTINE YawRateControl(avrSWAP, CntrPar, LocalVar, objInst, zmqVar, DebugVar,
         END IF
     END SUBROUTINE YawRateControl
+ !-------------------------------------------------------------------------------------------------------------------------------
+    
+     SUBROUTINE FeedForwardCPC(LocalVar, CntrPar, objInst, ErrVar)
+    ! Subroutine for calculating feedforward collective pitch commands using the first beam point only
+    
+        USE ROSCO_Types, ONLY : ControlParameters, LocalVariables, ObjectInstances, ErrorVariables
+        
+       ! REAL(C_FLOAT), INTENT(INOUT) :: avrSWAP(*)   ! The swap array, used to pass data to, and receive data from, the DLL controller.
+        TYPE(LocalVariables), INTENT(INOUT)          :: LocalVar
+        TYPE(ControlParameters), INTENT(INOUT)       :: CntrPar
+        TYPE(ObjectInstances), INTENT(INOUT)         :: objInst
+        TYPE(ErrorVariables), INTENT(INOUT)          :: ErrVar
+        INTEGER(4)                                   :: iBuffer
+        REAL(8)                                      :: ActionTime(LocalVar%MaxBufferStep_REWS)
+        INTEGER(4)                                   :: ArrayPosition
+   
+        
+        !--------- Shift the filter and Time in the buffer and assign new values -------------------------
+         
+       LocalVar%REWS_f(2:LocalVar%MaxBufferStep_REWS) = LocalVar%REWS_f(1:LocalVar%MaxBufferStep_REWS-1)
+       LocalVar%REWS_f_Time(2:LocalVar%MaxBufferStep_REWS) = LocalVar%REWS_f_Time(1:LocalVar%MaxBufferStep_REWS-1)
+       LocalVar%REWS_f(1)    = LPFilter(LocalVar%LidSpeed(1), LocalVar%DT, CntrPar%FF_LPFCornerFreq, LocalVar%FP, LocalVar%iStatus, .FALSE., objInst%instLPF)
+       LocalVar%LeadTime_1   =  (-LocalVar%MsrPosition(1,1))/LocalVar%REWS_f(1)
+       LocalVar%REWS_f_time(1) = LocalVar%LeadTime_1 + LocalVar%Time
+      
+           
+        !--------- Get the buffer ID that has the time closest to the pitch action time ------------------
+        
+       ActionTime   =  ABS(LocalVar%REWS_f_Time - LocalVar%Time - CntrPar%Pitch_PreviewTime)         ! This determines the time that needs to be matched to the wind speed record
+       ArrayPosition    = INT(MINLOC(ActionTime, 1))                                             ! This determines the position  in the REWS array that needs to be called
+       LocalVar%REWS_b  = LocalVar%REWS_f(ArrayPosition)                                          ! REWS_b is the wind speed AT the desired time
+       LocalVar%FF_Pitch = interp1d(CntrPar%REWS_curve,CntrPar%Pitch_curve, LocalVar%REWS_b, ErrVar)
+       LocalVar%FF_Pitch_Error  = LocalVar%FF_Pitch - (LocalVar%BlPitch(1)+ LocalVar%BlPitch(2)+ LocalVar%BlPitch(3))/3
+
+       
+       !---------- Limit feedforward pitch action depending on wind speed conditions ----------------------
+        
+        IF (LocalVar%URefLid > 15) THEN
+            
+             IF ((LocalVar%REWS_b <= 15) .AND. (LocalVar%REWS_b > 13) ) THEN
+                  LocalVar%FF_Pitch_Error = saturate(LocalVar%FF_Pitch_Error, -0.05, 0.15) 
```

Comment:
Please add a `_DbKi` suffix to all double-kind constants.  There are other examples in the code.